### PR TITLE
fix: kata-deploy.sh VERSION_ID unbound-variable

### LINKED ISSUE #9670: VERSION_ID: unbound variable error in kata-deploy:3.5.0
# Description of problem

kata-deploy:3.5.0 exits in error in rolling linux distros without VERSION_ID in os_release, e.g. archlinux, void linux

# Expected result

It should deploy kata-containers.

# Actual result

```text
 /opt/kata-artifacts/scripts/kata-deploy.sh: line 276: VERSION_ID: unbound variable
```

# Further information

kata-deploy:3.4.0 doesn't have this issue.

# Kata Containers survey

Please consider taking the survey to help us help you: https://openinfrafoundation.formstack.com/forms/kata_containers_user_survey

### PR DESCRIPTION
**Pull request by networkhermit**
fixes #9670

VERSION_ID is not guaranteed to be specified in os-release, this makes kaka-deploy breaks in rolling distros like arch linux and void linux.

> Note that operating system vendors may choose not to provide version information, for example to accommodate for rolling releases. In this case, VERSION and **VERSION_ID may be unset. Applications should not rely on these fields to be set.**

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
Oh!

Sorry for the breakage there!

We can cut a 3.5.1 once this one gets merged.

**Comment by fidencio** [comment]
@kata-containers/architecture-committee

**Comment by fidencio** [comment]
@networkhermit, you need to adapt your commit message.

Something like:
```
kata-deploy: Fix unbound VERSION_ID

VERSION_ID is not guaranteed to be specified in os-release, this makes kaka-deploy breaks in rolling distros like arch linux and void linux.

Note that operating system vendors may choose not to provide version information, for example to accommodate for rolling releases. In this case, VERSION and VERSION_ID may be unset. Applications should not rely on these fields to be set.

Signed-off-by: $YOUR_NAME <$YOUR_EMAIL>
```

**Review by stevenhorsman — APPROVED** [review]
LGTM once the commit message is updated.

**Comment by fidencio** [comment]
@networkhermit, could you rebae your PR atop of `main`?  That will help us to get the issues we're facing with AKS fixed.

**Comment by networkhermit** [comment]
> Oh!
> 
> Sorry for the breakage there!
> 
> We can cut a 3.5.1 once this one gets merged.

It's not urgent for me.

But if we want make a new release, is it possible to fix #9416? 

kata-deploy.yaml in 3.5.0 branch still use latest image.

https://github.com/kata-containers/kata-containers/blob/3.5.0/tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml#L20

https://github.com/kata-containers/kata-containers/blob/6a4ff08156f716db53def098e0506e43be9698c9/tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml#L20

**Comment by networkhermit** [comment]
> @networkhermit, could you rebae your PR atop of `main`? That will help us to get the issues we're facing with AKS fixed.

@fidencio I think it is already atop of main.

**Comment by fidencio** [comment]
> @fidencio I think it is already atop of main.

Okay, then we have some AKS issues again, let me fix those in a new PR.

**Comment by networkhermit** [comment]
> > Oh!
> > Sorry for the breakage there!
> > We can cut a 3.5.1 once this one gets merged.
> 
> It's not urgent for me.
> 
> But if we want to make a new release, is it possible to fix #9416?
> 
> kata-deploy.yaml in 3.5.0 branch still use latest image.
> 
> https://github.com/kata-containers/kata-containers/blob/3.5.0/tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml#L20
> 
> https://github.com/kata-containers/kata-containers/blob/6a4ff08156f716db53def098e0506e43be9698c9/tools/packaging/kata-deploy/kata-deploy/base/kata-deploy.yaml#L20

@fidencio @stevenhorsman Kindly ping if you missed the comment. If that behavior is not expected, I believe it is more important to fix the highly likely version drift.

**Comment by fidencio** [comment]
> @fidencio @stevenhorsman Kindly ping if you missed the comment. If that behavior is not expected, I believe it is more important to fix the highly likely version drift.

We have started a Slack discussion about this: https://katacontainers.slack.com/archives/C879ACQ00/p1716209200256559

**Pull request merged**
